pmlc/dialect/pxa/transforms/tile.cc

```diff
@@ -36,6 +36,16 @@ void performTiling(AffineParallelOp op, llvm::ArrayRef<int64_t> tileSizes) {
   auto &outerLoopOps = outerBody->getOperations();
   innerLoopOps.splice(std::prev(innerLoopOps.end()), outerLoopOps,
                       std::next(outerLoopOps.begin(), 1), outerLoopOps.end());
+  // Replace old indices with new indices
+  auto innerIdxs = inner.getBody()->getArguments();
+  for (unsigned i = 0; i < outerIdxs.size(); ++i) {
+    outerIdxs[i].replaceAllUsesWith(innerIdxs[i]);
+  }
+  unsigned numIdxs = inner.lowerBoundsMap().getNumInputs();
+  for (unsigned i = 0; i < numIdxs; ++i) {
+    inner.setOperand(i, outerIdxs[i]);
+    inner.setOperand(i + numIdxs, outerIdxs[i]);
+  }
```

Comment:
This looks great!  I'm slightly surprised the MLIR allows the inner AP to even temporarily have uses that are first defined in it's interior, but perhaps the check for recursive relations happens only during validation, and of course by that time it's been fixed.  Presuming that is the correct semantics (which if the debug build doesn't complain, I assume it is), that's a very clean solution!